clang/include/clang/Tooling/DependencyScanning/DependencyScanningTool.h

```diff
@@ -57,6 +57,10 @@ struct TranslationUnitDeps {
   /// determined that the differences are benign for this compilation.
   std::vector<ModuleID> ClangModuleDeps;
 
+  /// A list of module names that are visible to this translation unit. This
+  /// includes both direct and transitive module dependencies.
+  std::vector<std::string> VisibleModules;
```

Comment:
@cyndyishida , Swift queries `DependencyScanningTool::getModuleDependencies` with a `ModuleName` argument, which returns a `std::vector<ModuleDeps>`, not `TranslationUnitDeps`. 

Not sure what the best way to support that is, whether it is to add `VisibleModules` to the "main" `ModuleDeps` object of the module which was queried by-name or to change the return type of this API.

Reply:
FWIW, Swift does also use `DependencyScanningTool::getTranslationUnitDependencies` when scanning individual header files.